HHS/Head-Start-TTADP
# [TTAHUB-174]/update list of topics plus admin diagnostics table

### 174

## Description of change
 
-  Update topic names in the front end javascript constants in which they live
-  Write a recurring job to execute a sequelize update function to rename the offending topics
-  Added unit test for job

## How to test

Create a new activity report, go to the topics and resources pane

Confirm that the topics are in alphabetical order and that:

    "CLASS: Classroom Organization"
    "Behavioral/Mental Health / Trauma"
    "Curriculum (Instructional or Parenting)"
    "Environmental Health and Safety / EPRR"
    "Quality Improvement Plan/QIP"

appear.

To test recurring job aside from unit test, you'll need to run the updateTopicNames function and have some of the un-renamed topics in the database. As an example, to run it locally for me, I added it to an API handler so it would run arbitrarily, and ran SQL statements to confirm that the topics had been renamed.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-174


## Checklists

### Every PR


- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy


### After merge/deploy



### 96

## Description of change
Since the production db access is limited, this PR exposes one of the tables used for debugging. With the change the request errors table is visible and records can be deleted. The frontend component - react-admin provides a lot of functionality by default.

## How to test
- go to https://tta-smarthub-dev.app.cloud.gov/
- navigate to the admin UI
- verify that there is a button with "diag" on it
- click on the "diag" button
- verify that a Requesterrors displays without data in it
-

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-96


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
